File: tests/typesafety/python/printed_steps.py
# expect: passes
# title: The steps print their log, and the caller cannot read it
def double(n: int) -> int:
    print(f"doubled {n}")
    return n * 2


def add_tax(n: int) -> int:
    print(f"taxed {n}")
    return n + n // 10


def total(n: int) -> int:
    return add_tax(double(n))

File: tests/typesafety/typescript/printed_steps.ts
// expect: passes

function double(n: number): number {
  console.log(`doubled ${n}`);
  return n * 2;
}

function addTax(n: number): number {
  console.log(`taxed ${n}`);
  return n + Math.floor(n / 10);
}

export function total(n: number): number {
  return addTax(double(n));
}
